staging/src/k8s.io/kubectl/pkg/cmd/create/create_secret_test.go

```diff
@@ -482,40 +520,57 @@ func TestCreateSecretGeneric(t *testing.T) {
 				secret, err = secretOptions.createSecret()
 			}
 
-			if !test.expectErr && err != nil {
-				t.Errorf("test %s, unexpected error: %v", name, err)
+			if test.expectErr == "" && err != nil {
+				t.Errorf("unexpected error: %v", err)
+				return
 			}
-			if test.expectErr && err == nil {
-				t.Errorf("test %s was expecting an error but no error occurred", name)
+			if test.expectErr != "" && err == nil {
+				t.Errorf("was expecting an error but no error occurred")
+				return
```

Comment:
Added `return` because without it if the test fails it will throw a panic for the next line when we want to dereference the err `err.Error()` but the err will be `nil`. Logically, it makes sense to have `return` there because we want to move on to the next test cases if the current fail.

Reply:
Actually I think I didn't explain it well enough, 
For example if I have a test case
```go
		"foo": {
			secretName:  "foo",
			fromLiteral:    []string{"key1=value1"},
			expectErr:   `foo`,
		},
```
when it will first reach this line, which is fine because we are expecting an error and the err is `nil`
```go
if test.expectErr != "" && err == nil
```
But when it reaches the next line, `err.Error()` will cause a panic because `err` is `nil`.
```go
if test.expectErr != "" && test.expectErr != err.Error() {
```

If I do,
```go
if test.expectErr != "" {
  require.EqualError(t, test.expectErr, err.Error())
} 
require.NoError(t, err)
if !apiequality.Semantic.DeepEqual ... {
}

```

woudln't that cause panic too if err is `nil`?

Reply:
Ah yes, you're right. You'd need an additional `require.Error(t, err)` as the second line.

Reply:
If that's the case, do you think `require` is better or `return`? Personally, I think `return` would make the code cleaner.

Reply:
I find the `require` version more succinct/readable, but it's a personal preference. I'll LGTM either way once the error wording fix is pushed.

Reply:
sure! let's go with `require` then, I will play around with it and make an update. Thank you